x-pack/examples/third_party_vis_lens_example/public/plugin.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ExpressionsSetup } from 'src/plugins/expressions/public';
+import { FieldFormatsStart } from 'src/plugins/field_formats/public';
+import { Plugin, CoreSetup, AppNavLinkStatus } from '../../../../src/core/public';
+import { DataViewsPublicPluginStart, DataView } from '../../../../src/plugins/data_views/public';
+import { LensPublicSetup, LensPublicStart } from '../../../plugins/lens/public';
+import { DeveloperExamplesSetup } from '../../../../examples/developer_examples/public';
+import { TypedLensByValueInput, PersistedIndexPatternLayer } from '../../../plugins/lens/public';
+import { getRotatingNumberRenderer, rotatingNumberFunction } from './expression';
+import { getRotatingNumberVisualization } from './visualization';
+import { RotatingNumberState } from './types';
```

Comment:
seems this plugin is providing lens plugin with some state. lens doesn't own that state so you should implement persistable state item interface on your lens visualizations, so 3rd party plugins registering additional visualizations have a way to migrate that state later.

Reply:
Great catch, thanks - gonna extend the API